game/platform.hh

```diff
@@ -21,9 +21,9 @@ using STAT = struct stat;
 #endif
 
 struct Platform {
-enum platforms { windows, linux, macos, bsd, solaris, unix };
+enum HostOS { PERFORMOUS_WIN, PERFORMOUS_LINUX, PERFORMOUS_MAC, PERFORMOUS_BSD, PERFORMOUS_SOLARIS, PERFORMOUS_UNIX };
```

Comment:
Within c++ there's two ways to declare an enum:
1. Plain old enum like shown
2. an Enum class 
The latter is used more and should be preferred because they cause fewer surprises that could potentially lead to bugs. It also makes refactoring way easier.
And as third we already use enum class with the a lot of our enums. Granted not yet all use it.

An example copied over from [stackoverflow](https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum) on the usage:
```
enum Color { red, green, blue };                    // plain enum 
enum Card { red_card, green_card, yellow_card };    // another plain enum 
enum class Animal { dog, deer, cat, bird, human };  // enum class
enum class Mammal { kangaroo, deer, human };        // another enum class

void fun() {

    // examples of bad use of plain enums:
    Color color = Color::red;
    Card card = Card::green_card;

    int num = color;    // no problem

    if (color == Card::red_card) // no problem (bad)
        cout << "bad" << endl;

    if (card == Color::green)   // no problem (bad)
        cout << "bad" << endl;

    // examples of good use of enum classes (safe)
    Animal a = Animal::deer;
    Mammal m = Mammal::deer;

    int num2 = a;   // error
    if (m == a)         // error (good)
        cout << "bad" << endl;

    if (a == Mammal::deer) // error (good)
        cout << "bad" << endl;

}
```

If we actually wanna keep on using the plain enums we should be using typed enums: `enum Animal: unsigned int {dog, deer, cat, bird}` introduced in c++11

Reply:
@Baklap4 Yes, this is better and avoids the ugly names too.

Reply:
Just checking: with 'this' you mean the `enum class` variant right?

Reply:
Yes, enum class and short lower-case labels.

Reply:
This was taken from a larger PR. I originally had used enum class (albeit with similarly long names) and @Tronic said they were too long.

I went for long, specific names for one specific circumstance (although there could be others), using gcc with `-std=gnu[11|14|17] treats `linux` as a macro and all sort of shit breaks.

Reply:
Right, cannot avoid macro expansion with namespacing. Oh, well. Anyway, the names can be more reasonable still. Cannot change all naming to use very unique names just to avoid macro collisions.

For instance, `<windows.h>` defines a macro named `max` which obviously spoils your efforts at using `std::max(...)`. Clean up the excess macros when you can (`-D WIN32_LEAN_AND_MEAN --std=c++17`).

Reply:
Yeah; in general MXE at least includes most things using `-D WIN32_LEAN_AND_MEAN`and many libraries also have also try to get around it by specifically checking for `max` and `min` which are particularly troublesome.

How about keeping it unscoped and changing to `OS_WIN, OS_MAC`and so on? I'm using caps mostly because enums elsewhere in the code are also using caps. I should probably also change `HostOS`to `hostOS`, though.